src/Hazelcast.Net/CP/FencedLock.cs

```diff
@@ -114,8 +100,18 @@ public async Task<long> GetFenceAsync()
         /// <inheritdoc/>  
         public async Task<int> GetLockCountAsync()
         {
-            var contextId = AsyncContext.Current.Id; // the asynchronous context identified (equivalent to Java thread identifier)
-            using var contextLock = await _contextLocker.LockAsync(contextId).CfAwait(); // lock the asynchronous context
+            // the original Java code (below) does some sanity-checking on this operation
```

Comment:
Can we please make sync with Java and if there is a valid argument for improvement, let's also change the Java side as well. I can not reason why we do not need the checks in Java. Can you please elaborate why it is not required?

Reply:
This `GetLockCountAsync` method gets the number of times the lock has been locked, it would be zero if the lock is not locked, 1 if locked once, 2 if re-entered once, etc. It has nothing to do with the current thread / context. Same happens with `IsLockedAsync` method that determines whether the lock has been locked by any thread, regardless of the current thread / context.

Yet the Java code uses the opportunity to do some cleanup and sanity check. For instance, if the lock is not owned by the current thread, it verifies that the current thread does not *think* it owns the lock, else it throws a `LockOwnershipLostException`.

This is rather bizare: you just want to know whether the lock is locked, or how many times it is locked, and you can get an exception because you lost the lock. IMHO these are totally different concerns, it is confusing and makes no sense.

But I understand the need to align with Java. It's quite easy to do and I kinda expected you to ask ;) Just let me know what you prefer.

Reply:
> This is rather bizare: you just want to know whether the lock is locked, or how many times it is locked, and you can get an exception because you lost the lock. IMHO these are totally different concerns, it is confusing and makes no sense.

Feeling same.

Reply:
Let's talk. i expect all the clients behave the same and fix all or none.

Reply:
My preference: Sync with Java or otherwise fix Java as well.

Reply:
wont fix as: it makes no sense to add an explicit context parameter to the `IsLocked` method that does not use the value of the parameter + does *not* accept such a parameter in Java, and we cannot "fix" Java as Java and .NET have to be different here due to the difference in threading / context model. See comments in the code for details.